client/components/modal-addtrip.jsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+
+export default class TripModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      tripName: '',
+      tripDestination: '',
+      departureDate: '',
+      returnDate: '',
+      numberOfDays: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleClickOff = this.handleClickOff.bind(this);
+    this.calculateDays = this.calculateDays.bind(this);
+  }
+
+  calculateDays() {
+    const date1 = Date.parse(this.state.departureDate);
+    const date2 = Date.parse(this.state.returnDate);
+    const timeDiff = date2 - date1;
+    let days = Math.floor(timeDiff / (1000 * 60 * 60 * 24));
+    days += 1;
+    this.setState(state => ({
+      numberOfDays: days
+    }));
+  }
+
+  handleChange(event) {
+    const value = event.target.value;
+    this.setState({
+      [event.target.name]: value
+    }, this.calculateDays);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const req = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(this.state)
+    };
+    fetch('/api/trip', req)
+      .then(response => response.json())
+      .then(newTrip => {
+        const newTripEntriesArray = this.props.tripEntries.concat(newTrip);
+        this.props.homeState({ tripEntries: newTripEntriesArray });
+        event.target.reset();
+      })
+      .catch(err => console.error(err));
+
+    this.props.homeState({
+      modalView: false
+    });
+  }
+
+  handleClickOff(e) {
+    this.props.homeState({
+      modalView: false
+    });
+  }
+
+  stopPropagation(e) {
+    e.stopPropagation();
+  }
+
+  render() {
+    const today = new Date();
+    let month = today.getMonth() + 1;
+    let day = today.getDate() + 1;
+    const year = today.getFullYear();
+    if (month < 10) {
+      month = '0' + month.toString();
+    }
+    if (day < 10) {
+      day = '0' + day.toString();
+    }
+    const maxDate = `${year}-${month}-${day}`;
```

Comment:
I noticed this is in another component too. I recommend moving a helper function into its own file so you can import it into each component file.